app/src/main/java/de/westnordost/streetcomplete/quests/building_type/AddBuildingTypeForm.java

```diff
@@ -0,0 +1,168 @@
+			MOSQUE =		new Item("mosque", R.drawable.building_mosque,	R.string.quest_buildingType_mosque),
+			TEMPLE =		new Item("temple", R.drawable.building_temple,	R.string.quest_buildingType_temple),
+			SYNAGOGUE =		new Item("synagogue", R.drawable.building_synagogue,	R.string.quest_buildingType_synagogue),
+
+			CARPORT =		new Item("carport", R.drawable.building_carport, R.string.quest_buildingType_carport),
+			GARAGE =		new Item("garage",	R.drawable.building_garage, R.string.quest_buildingType_garage),
+			GARAGES =		new Item("garages", R.drawable.building_garages, R.string.quest_buildingType_garages),
+			PARKING = 		new Item("parking", R.drawable.building_parking, R.string.quest_buildingType_parking),
+
+			FARM =			new Item("farm", R.drawable.building_farm, R.string.quest_buildingType_farmhouse),
+			FARM_AUXILIARY = new Item("farm_auxiliary", R.drawable.building_farm_auxiliary, R.string.quest_buildingType_farm_auxiliary),
+			BARN =			new Item("barn", R.drawable.building_barn,	R.string.quest_buildingType_barn),
+
+			SHED =			new Item("shed", R.drawable.building_shed,	R.string.quest_buildingType_shed),
+			HUT =			new Item("hut", R.drawable.building_hut, R.string.quest_buildingType_hut),
+			ROOF =			new Item("roof", R.drawable.building_roof, R.string.quest_buildingType_roof),
+			GREENHOUSE =	new Item("greenhouse",	R.drawable.building_greenhouse,	R.string.quest_buildingType_greenhouse),
+			CABIN =			new Item("cabin", R.drawable.building_cabin, R.string.quest_buildingType_cabin),
+			CONSTRUCTION =	new Item("construction", R.drawable.building_construction, R.string.quest_buildingType_construction),
+			RUINS =			new Item("ruins", R.drawable.building_ruins, R.string.quest_buildingType_ruins),
+			SERVICE =		new Item("service", R.drawable.building_service, R.string.quest_buildingType_service);
+
+	private final Item[] TOP_BUILDINGS = new Item[] {
+			HOUSE, DETACHED, APARTMENTS,
+			GARAGE, GARAGES, HUT,
+			RETAIL, COMMERCIAL, INDUSTRIAL
+	};
+
+	private final Item[] ALL_BUILDINGS = new Item[] {
+			new Item("residential", R.drawable.panorama_building_residential, R.string.quest_buildingType_residential, new Item[]{
+					DETACHED, HOUSE, TERRACE, APARTMENTS, BUNGALOW,
+					FARM, HOUSEBOAT, STATIC_CARAVAN, DORMITORY, HOTEL,
+			}),
+			new Item(null, R.drawable.panorama_building_commercial, R.string.quest_buildingType_commercial, new Item[]{
+					COMMERCIAL, INDUSTRIAL, RETAIL, KIOSK, WAREHOUSE
+			}),
+			new Item("civic", R.drawable.panorama_building_civic, R.string.quest_buildingType_civic, new Item[]{
+					COLLEGE, UNIVERSITY, SCHOOL, PUBLIC, HOSPITAL, STADIUM, TRAIN_STATION, TRANSPORTATION
+			}),
+			new Item("religious", R.drawable.panorama_building_religious, R.string.quest_buildingType_religious, new Item[]{
+					CHURCH, CATHEDRAL, CHAPEL, MOSQUE, TEMPLE, SYNAGOGUE
+			}),
+			new Item(null, R.drawable.panorama_building_cars, R.string.quest_buildingType_cars, new Item[]{
```

Comment:
What about a petrol station? I admit that building=petrol_station is rare but I think that it is optimal. Optionally, give an option titled "petrol station" that sets building=shop.

Reply:
Might be okay, but the actual building should be tagged with [`building=roof`](https://wiki.openstreetmap.org/wiki/Tag:building=roof) AFAIK... Shouldn't it?

Reply:
Ops, it is more complicated.

There may be

- roof (obviously tagged as building=roof)
- petrol station building (usually with place to pay for petrol, shop, toilets, sometimes also with fast food)
- building mistagged and containing both

Maybe have "roof of petrol station" and "petrol station building" in "things for cars"?

Reply:
Perhaps it'd help to change the string "For cars" to "For car keeping" or "Car parking" and add Roof to the Commercial category (so it'd be in twice)? The petrol station building should be Retail (usually also Convenience Shop) or Kiosk, right?

Reply:
@rugk What do you think about the solution proposed by @matkoniecz?